Make suitable to ruby-ng.eclass
https://bugs.gentoo.org/320617

Support >=dev-ruby/bsearch-1.5.0
https://bugs.gentoo.org/455258


--- migemo-0.40.orig/Makefile.am
+++ migemo-0.40/Makefile.am
@@ -15,17 +15,17 @@
 	sed -e 's!@pkgdatadir@!$(pkgdatadir)!g' migemo.el.in > $@
 
 migemo-dict: # SKK-JISYO.L
-	ruby -I. migemo-convert.rb SKK-JISYO.L > $@
+	$(RUBY) -I. migemo-convert.rb SKK-JISYO.L > $@
 
 migemo-dict.idx: migemo-dict
-	ruby -I. migemo-index.rb migemo-dict > migemo-dict.idx
+	$(RUBY) -I. migemo-index.rb migemo-dict > migemo-dict.idx
 
 migemo-dict.cache: frequent-chars migemo-dict migemo-dict.idx
-	sort frequent-chars | ruby -I. migemo-cache.rb migemo-dict
+	sort frequent-chars | $(RUBY) -I. migemo-cache.rb migemo-dict
 
 frequent-chars: genchars.sh migemo-dict
-	sh genchars.sh > tmp.list1
-	cat tmp.list1 | ruby -rromkan -ne 'puts $$_.to_kunrei' > tmp.list2
+	RUBY=$(RUBY) sh genchars.sh > tmp.list1
+	cat tmp.list1 | $(RUBY) -rromkan -ne 'puts $$_.to_kunrei' > tmp.list2
 	cat tmp.list1 tmp.list2 | sort | uniq > frequent-chars
 
 clean-local:
--- migemo-0.40.orig/acinclude.m4
+++ migemo-0.40/acinclude.m4
@@ -123,7 +123,7 @@
   AC_MSG_CHECKING([where .rb files should go])
   if test "x$rubydir" = x; then
     changequote(<<, >>)
-    rubydir=`ruby -rrbconfig -e 'puts Config::CONFIG["sitedir"]'`
+    rubydir=`$RUBY -rrbconfig -e 'puts Config::CONFIG["sitelibdir"]'`
     changequote([, ])
   fi
   AC_MSG_RESULT($rubydir)
--- migemo-0.40.orig/configure.in
+++ migemo-0.40/configure.in
@@ -10,7 +10,9 @@
 AM_PATH_RUBYDIR
 
 echo -n "checking Ruby/Bsearch... "
-if ruby -rbsearch -e 'exit(if Bsearch::VERSION >= "1.2" then 0 else 1 end)'; then
+if $RUBY -e \
+   'require "bsearch"
+    exit(if Bsearch::VERSION::STRING >= "1.2" then 0 else 1 end)'; then
    echo found
 else
    echo not found
@@ -19,7 +21,7 @@
 fi
 
 echo -n "checking Ruby/Romkan... "
-if ruby -rromkan -e 'exit(if Romkan::VERSION >= "0.3" then 0 else 1 end)'; then
+if $RUBY -rromkan -e 'exit(if Romkan::VERSION >= "0.3" then 0 else 1 end)'; then
    echo found
 else
    echo not found
--- migemo-0.40.orig/genchars.sh
+++ migemo-0.40/genchars.sh
@@ -1,6 +1,8 @@
 #! /bin/sh
 
-ruby -rromkan -nle 'head = split[0]; if /^\w+$/ =~ head then puts head else roma = head.to_roma; puts roma, roma.to_kunrei end' migemo-dict |uniq> tmp.ascii.words
+RUBY=${RUBY:-ruby}
+
+${RUBY} -rromkan -nle 'head = split[0]; if /^\w+$/ =~ head then puts head else roma = head.to_roma; puts roma, roma.to_kunrei end' migemo-dict |uniq> tmp.ascii.words
 
 # Get the top 500 frequent ngrams.
 for i in 1 2 3 4 5 6 7 8; do
--- migemo-0.40.orig/tests/Makefile.am
+++ migemo-0.40/tests/Makefile.am
@@ -5,13 +5,13 @@
 
 
 test-dict.idx: test-dict ../migemo-index.rb
-	ruby -I.. ../migemo-index.rb test-dict > test-dict.idx
+	$(RUBY) -I.. ../migemo-index.rb test-dict > test-dict.idx
 
 test-dict.cache: test-dict test-dict.idx ../migemo-cache.rb
-	ruby -rromkan -ne 'puts $$1.to_roma if /^(.+?)	/' test-dict |\
+	$(RUBY) -rromkan -ne 'puts $$1.to_roma if /^(.+?)	/' test-dict |\
 	while read line; do\
-	    echo $$line | ruby -ne 'chomp!;1.upto($$_.length) do |x| puts $$_[0,x] end';\
-	done | ruby -I.. ../migemo-cache.rb test-dict
+	    echo $$line | $(RUBY) -ne 'chomp!;1.upto($$_.length) do |x| puts $$_[0,x] end';\
+	done | $(RUBY) -I.. ../migemo-cache.rb test-dict
 
 clean-local:
 	rm -f tmp.*
